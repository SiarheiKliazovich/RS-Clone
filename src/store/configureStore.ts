/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
