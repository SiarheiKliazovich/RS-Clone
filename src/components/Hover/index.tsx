import { cloneElement, FunctionComponent, useState } from "react";
import { useSelector } from "react-redux";
import { HoverType } from "../../types";
import { IStore } from "../../interfaces/store";

const Hover: FunctionComponent<HoverType> = ({ children }: HoverType) => {
  const { accentColor } = useSelector((state: IStore) => state.appInterface);

  const [isHovered, setIsHovered] = useState(false);

  return (
    <>
      {cloneElement(children, {
        style: {
          backgroundColor: isHovered ? accentColor.hover : accentColor.static,
        },
        onMouseOver: () => setIsHovered(true),
        onMouseOut: () => setIsHovered(false),
      })}
    </>
  );
};

export default Hover;