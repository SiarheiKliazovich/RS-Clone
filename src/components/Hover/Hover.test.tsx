import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Hover from ".";

const mockStore = configureStore([]);

describe("Hover", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Hover>
            <div>Content</div>
          </Hover>
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
