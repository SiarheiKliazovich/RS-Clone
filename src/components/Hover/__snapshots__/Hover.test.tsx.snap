// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hover renders correctly 1`] = `
<div
  style="background-color: rgb(0, 0, 0);"
>
  Content
</div>
`;
